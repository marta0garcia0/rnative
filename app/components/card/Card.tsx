import { StyleSheet, TouchableOpacity } from "react-native";

import { ThemedText } from "@/components/ThemedText";
import { ThemedView } from "@/components/ThemedView";
import { User } from "@/app/api/models";

export function Card({ user, onSelect }: { user: User; onSelect: () => void }) {
  return (
    <ThemedView>
      <TouchableOpacity
        onPress={onSelect}
        style={styles.container}
        activeOpacity={0.8}
        testID={"cardId"}
      >
        <ThemedText type="defaultSemiBold">{user.name}</ThemedText>
        <ThemedText>{user.username}</ThemedText>
      </TouchableOpacity>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    padding: 14,
    alignItems: "flex-start",
    gap: 6,
    borderWidth: 1,
    borderColor: "grey",
  },
});
