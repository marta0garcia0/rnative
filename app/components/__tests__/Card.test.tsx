import * as React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";

import { Card } from "../card/Card";

describe("Card component", () => {
  it(`renders correctly`, () => {
    const onSelect = jest.fn();
    const tree = renderer
      .create(
        <Card
          user={{
            id: 1,
            name: "John Doe",
            username: "johndoe",
            email: "johndoe@example.com",
            address: {
              street: "123 Elm St",
              suite: "Apt 4",
              city: "Gotham",
              zipcode: "12345",
              geo: {
                lat: "40.7128",
                lng: "-74.0060",
              },
            },
            phone: "555-555-5555",
            website: "johndoe.com",
            company: {
              name: "Doe Enterprises",
              catchPhrase: "We do things differently",
              bs: "innovate synergy",
            },
          }}
          onSelect={onSelect}
        ></Card>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("display name and surname correctly", () => {
    const onSelect = jest.fn();
    const { getByTestId, debug, getByText } = render(
      <Card
        user={{
          id: 1,
          name: "John Doe",
          username: "johndoe",
          email: "johndoe@example.com",
          address: {
            street: "123 Elm St",
            suite: "Apt 4",
            city: "Gotham",
            zipcode: "12345",
            geo: {
              lat: "40.7128",
              lng: "-74.0060",
            },
          },
          phone: "555-555-5555",
          website: "johndoe.com",
          company: {
            name: "Doe Enterprises",
            catchPhrase: "We do things differently",
            bs: "innovate synergy",
          },
        }}
        onSelect={onSelect}
      />,
    );
    debug();
    const buttonComponent = getByTestId("cardId");
    fireEvent(buttonComponent, "press");
    const name = getByText("John Doe");
    const username = getByText("johndoe");

    expect(onSelect).toHaveBeenCalled();
    expect(name).toBeDefined();
    expect(username).toBeDefined();
  });
});
