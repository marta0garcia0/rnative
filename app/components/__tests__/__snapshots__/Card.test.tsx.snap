// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "borderColor": "grey",
        "borderRadius": 4,
        "borderWidth": 1,
        "gap": 6,
        "opacity": 1,
        "padding": 14,
      }
    }
    testID="cardId"
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      John Doe
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      johndoe
    </Text>
  </View>
</View>
`;
